src/libstore/build.cc

```diff
@@ -1906,6 +1941,9 @@ void DerivationGoal::buildDone()
     done(BuildResult::Built);
 }
 
+void DerivationGoal::resolvedFinished() {
```

Comment:
This function seems a bit unnecessary.

Reply:
Well many derivation goals could resolve to the same derivation, Trying to make the resolve derivations track all those to call `done` on seems wrong to me.